src/react/utils.js

```diff
@@ -962,40 +984,116 @@ function applyClonedTemporalAlias(realm: Realm, props: ObjectValue, clonedProps:
     { skipInvariant: true }
   );
   invariant(temporalTo instanceof AbstractObjectValue);
-  invariant(clonedProps instanceof ObjectValue);
-  temporalTo.values = new ValuesDomain(clonedProps);
-  clonedProps.temporalAlias = temporalTo;
+  invariant(clonedObj instanceof ObjectValue);
+  temporalTo.values = new ValuesDomain(clonedObj);
+  clonedObj.temporalAlias = temporalTo;
   // Store the args for the temporal so we can easily clone
   // and reconstruct the temporal at another point, rather than
   // mutate the existing temporal
   realm.temporalAliasArgs.set(temporalTo, newTemporalArgs);
 }
 
-export function cloneProps(realm: Realm, props: ObjectValue, newChildren?: Value): ObjectValue {
-  let clonedProps = new ObjectValue(realm, realm.intrinsics.ObjectPrototype);
-
-  for (let [propName, binding] of props.properties) {
-    if (binding && binding.descriptor && binding.descriptor.enumerable) {
-      if (newChildren !== undefined && propName === "children") {
-        Properties.Set(realm, clonedProps, propName, newChildren, true);
-      } else {
-        Properties.Set(realm, clonedProps, propName, getProperty(realm, props, propName), true);
+export function cloneObject(
```

Comment:
This name is pretty generic. What does this "clone" exactly? What objects does it work on? What is its purpose?

My impression is this isn't just producing a simple shallow clone but has a more specific purpose, so might want to reflect this in naming.